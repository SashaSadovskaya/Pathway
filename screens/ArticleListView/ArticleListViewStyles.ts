import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  titleContainer: {
    marginVertical: 10,
    marginHorizontal: 20,
    justifyContent: 'center',
    alignItems: 'center',
    paddingBottom: 20,
    borderBottomWidth: 1,
    borderColor: 'grey',
  },
  title: {
    fontWeight: '800',
    marginVertical: 10,
  },
  image: {
    height: 150,
    width: 250,
  },
});
