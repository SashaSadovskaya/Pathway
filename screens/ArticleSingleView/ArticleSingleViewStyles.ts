import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    marginHorizontal: 20,
    justifyContent: 'center',
    alignContent: 'center',
  },
  title: {
    marginVertical: 20,
    alignSelf: 'center',
    fontSize: 16,
    fontWeight: '700',
  },
  image: {
    alignSelf: 'center',
    height: 200,
    width: 300,
    marginVertical: 20,
  },
  date: {
    alignSelf: 'center',
  },
});
