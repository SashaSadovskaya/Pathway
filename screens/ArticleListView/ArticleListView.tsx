import {useNavigation} from '@react-navigation/native';
import React, {useEffect, useState} from 'react';
import {styles} from './ArticleListViewStyles';
import {
  SafeAreaView,
  FlatList,
  Text,
  TouchableOpacity,
  Image,
} from 'react-native';
import {fetchArticles} from '../../utils';
import {Article} from '../../types';

const ArticleListView = () => {
  const navigation = useNavigation();
  const [articles, setArticles] = useState<Article[]>([]);

  useEffect(() => {
    (async () => {
      const response = await fetchArticles();

      setArticles(response);
    })();
  }, []);

  const renderItem = ({item}: {item: Article}) => {
    return (
      <TouchableOpacity
        key={item.id}
        activeOpacity={0.7}
        style={styles.titleContainer}
        onPress={() => {
          navigation.navigate('ArticleSingleView', {
            title: item.title,
            date: item.date,
            content: item.content,
            thumbnail: item.thumbnail,
          });
        }}>
        <Text style={styles.title}>{item.title}</Text>
        <Text>{item.date}</Text>
        <Text>{item.excerpt}</Text>
        <Image source={{uri: item.thumbnail}} style={styles.image} />
      </TouchableOpacity>
    );
  };

  return (
    <SafeAreaView style={styles.container}>
      <FlatList
        data={articles}
        renderItem={renderItem}
        keyExtractor={item => item.id.toString()}
      />
    </SafeAreaView>
  );
};

export default ArticleListView;
