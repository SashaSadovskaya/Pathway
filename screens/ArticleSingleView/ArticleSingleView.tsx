import React from 'react';
import {SafeAreaView, ScrollView, Text, View, Image} from 'react-native';
import {RouteProp} from '@react-navigation/native';
import {RootStackParamList} from '../../types';
import {styles} from './ArticleSingleViewStyles';

type ArticleSingleViewProps = {
  route: RouteProp<RootStackParamList, 'ArticleSingleView'>;
};

const ArticleSingleView = ({route}: ArticleSingleViewProps) => {
  const {title, date, content, thumbnail} = route.params;

  return (
    <SafeAreaView>
      <ScrollView>
        <View style={styles.container}>
          <Text style={styles.title} testID="title">
            {title}
          </Text>
          <Text style={styles.date} testID="date">
            {date}
          </Text>
          <Image source={{uri: thumbnail}} style={styles.image} />
          <Text testID="content">{content}</Text>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

export default ArticleSingleView;
