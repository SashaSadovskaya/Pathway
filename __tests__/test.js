import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';
import ArticleSingleView from '../screens/ArticleSingleView/ArticleSingleView';
import ArticleListView from '../screens/ArticleListView/ArticleListView';

jest.mock('@react-navigation/native');

describe('ArticleSingleView tests', () => {
  const title = 'test title';
  const date = 'test date';
  const content = 'test content';

  it('should display correct text', () => {
    const {getByTestId} = render(
      <ArticleSingleView
        route={{
          params: {
            title,
            date,
            content,
          },
        }}
      />,
    );

    const titleText = getByTestId('title').props.children;
    const dateText = getByTestId('date').props.children;
    const contentText = getByTestId('content').props.children;

    expect(titleText).toEqual(title);
    expect(dateText).toEqual(date);
    expect(contentText).toEqual(content);
  });

  it('should render correctly', () => {
    const component = render(
      <ArticleSingleView
        route={{
          params: {
            title,
            date,
            content,
          },
        }}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});

describe('ArticleListView tests', () => {
  it('should render correctly', () => {
    const component = render(<ArticleListView />);

    expect(component).toMatchSnapshot();
  });
});
