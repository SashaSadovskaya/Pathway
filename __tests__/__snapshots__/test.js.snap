// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListView tests should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`ArticleSingleView tests should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView>
    <View>
      <View>
        <Text
          testID="title"
        >
          test title
        </Text>
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "height": 200,
              "width": 300,
            }
          }
        />
        <Text
          testID="date"
        >
          test date
        </Text>
        <Text
          testID="content"
        >
          test content
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
